drivers/auxiliary/ipx800.h

```diff
@@ -0,0 +1,224 @@
+       SetR = 1 << 2,
+       ClearR = 1 << 3
+   } ;
+       
+	///////////////////////////////////////////
+	// IPX800 Communication
+	///////////////////////////////////////////
+	bool updateIPXData();
+    void updateObsStatus();
+    bool readCommand(IPX800_command);
+    bool writeCommand(IPX800_command, int toSet);
+    bool checkAnswer();
+    void readAnswer();
+    void recordData(IPX800_command command);
+    bool writeTCP(std::string toSend);
+	bool firstFonctionTabInit();
+	
+    virtual bool UpdateDigitalInputs() override;
+    virtual bool UpdateAnalogInputs() override; // IPX800 Analog Inputs not managed
+    virtual bool UpdateDigitalOutputs() override;
+    virtual bool CommandOutput(uint32_t index, OutputState command) override;
+	virtual bool saveConfigItems(FILE *fp) override;
+	
+  private:
+	bool roofPowerManagement = false;
+	Connection::TCP *tcpConnection {nullptr};
+	
+	// TO manage Password in a next release
+	//IText* getMyLogin();
+    //ITextVectorProperty getMyLoginVector();
+    
+	ISwitchVectorProperty getMyRelayVector(int i);
+    ISwitchVectorProperty getMyDigitsVector(int i);
+	
+	// List of possible commands 
+    enum {
+        UNUSED_RELAY,
+        ROOF_ENGINE_POWER_SUPPLY,
+        TUBE_VENTILATION,
+        HEATING_RESISTOR_1,
+        HEATING_RESISTOR_2,
+        ROOF_CONTROL_COMMAND,
+        MOUNT_POWER_SUPPLY,
+        CAM_POWER_SUPPLY,
+        OTHER_POWER_SUPPLY_1,
+        OTHER_POWER_SUPPLY_2,
+        OTHER_POWER_SUPPLY_3
+        } IPXRelaysCommands;
+
+	// List of possibles digital inputs
+    enum  {
+        UNUSED_DIGIT,
+        DEC_AXIS_PARKED,
+        RA_AXIS_PARKED,
+        ROOF_OPENED,
+        ROOF_CLOSED,
+        ROOF_ENGINE_POWERED,
+        RASPBERRY_SUPPLIED,
+        MAIN_PC_SUPPLIED,
+        OTHER_DIGITAL_1,
+        OTHER_DIGITAL_2 } IPXDigitalRead;
+	
+    char tmpAnswer[8]= {0};
+    bool setupParams();
+    float CalcTimeLeft(timeval);
+
+    ISState fullOpenLimitSwitch { ISS_ON };
+    ISState fullClosedLimitSwitch { ISS_OFF };
+    double MotionRequest { 0 };
+    struct timeval MotionStart { 0, 0 };
+
+    ISwitch RelaisInfoS[11] {};
+    ISwitch Relais1InfoS[11], Relais2InfoS[11],Relais3InfoS[11],Relais4InfoS[11],Relais5InfoS[11],Relais6InfoS[11],Relais7InfoS[11],Relais8InfoS[11] {};
+    ISwitchVectorProperty RelaisInfoSP[8] {};
+
+    ISwitch DigitalInputS[10] {};
+    ISwitch Digital1InputS[10], Digital2InputS[10], Digital3InputS[10], Digital4InputS[10], Digital5InputS[10], Digital6InputS[10], Digital7InputS[10], Digital8InputS[10];
+    ISwitchVectorProperty DigitalInputSP[8] {};
+
+    ISwitch Relay1StateS[2],Relay2StateS[2],Relay3StateS[2],Relay4StateS[2],Relay5StateS[2],Relay6StateS[2],Relay7StateS[2],Relay8StateS[2] ;
+    ISwitchVectorProperty RelaysStatesSP[8] ;
+    ISwitch Digit1StateS[2],Digit2StateS[2],Digit3StateS[2],Digit4StateS[2],Digit5StateS[2],Digit6StateS[2],Digit7StateS[2],Digit8StateS[2] ;
+    ISwitchVectorProperty DigitsStatesSP[8] ;
```

Comment:
What is this for? We already have the input/output properties defined by the interface, no need to define other properties unless there is a good reason.

Reply:
I understand your comment. But the ipx800 driver provide additionnal functions that Dome drivers can't provide. Those Inputs/Outputs allows the user to connect functions to Inputs/Outputs...
Those additionnal functions allows for exemple to lock roof moves to physical mount park position.

Reply:
I understand, but they don't belong in *this* driver. We can have another ROR or Dome driver that uses these functionalities. The purpose of IPX800 driver is to provide a **generic** Input/Ouput driver. There is nothing inherit in the IPX800 that is tied to domes or roll-offs.